File: src/main/java/net/zargor/afterlife/server/MimeTypes.java
package net.zargor.afterlife.server;

import lombok.Getter;

/**
 * The requested media-type. @see https://en.wikipedia.org/wiki/Media_type
 */
public enum MimeTypes {
	HTML("text/html;charset=utf-8", new String[]{".html", ".htm", ".shtml"}),
	PLAIN("text/plain;charset=utf-8", new String[]{".txt"}),
	CSS("text/css;charset=utf-8", new String[]{".css"}),
	JAVASCRIPT("text/javascript;charset=utf-8", new String[]{".js"}),
	XML("text/xml;charset=utf-8", new String[]{".xml"}),
	PNG("image/png", new String[]{".png"}),
	JPEG("image/jpeg", new String[]{".jpg", ".jpeg", ".jpe"}),
	GIF("image/gif", new String[]{".gif"}),
	ICO("image/x-icon", new String[]{".ico"}),
	WAV("audio/x-wav", new String[]{".wav"}),
	MP3("audio/mpeg", new String[]{".mp3"}),
	MP4("video/mp4", new String[]{".mp4"}),
	AVI("video/x-msvideo", new String[]{".avi"});

	@Getter
	private String mimeText;
	@Getter
	private String[] ending;

	MimeTypes(String mimeText, String[] ending) {
		this.mimeText = mimeText;
		this.ending = ending;
	}
}

File: src/main/kotlin/net/zargor/afterlife/web/MimeTypes.kt
package net.zargor.afterlife.web

/**
 * The requested media-type. @see https://en.wikipedia.org/wiki/Media_type
 */
enum class MimeTypes(val mimeText: String, val endings: Array<String>) {
    HTML("text/html;charset=utf-8", arrayOf(".html", ".htm", ".shtml")),
    PLAIN("text/plain;charset=utf-8", arrayOf(".txt")),
    CSS("text/css;charset=utf-8", arrayOf(".css")),
    JAVASCRIPT("text/javascript;charset=utf-8", arrayOf(".js")),
    XML("text/xml;charset=utf-8", arrayOf(".xml")),
    PNG("image/png", arrayOf(".png")),
    JPEG("image/jpeg", arrayOf(".jpg", ".jpeg", ".jpe")),
    GIF("image/gif", arrayOf(".gif")),
    ICO("image/x-icon", arrayOf(".ico")),
    WAV("audio/x-wav", arrayOf(".wav")),
    MP3("audio/mpeg", arrayOf(".mp3")),
    MP4("video/mp4", arrayOf(".mp4")),
    AVI("video/x-msvideo", arrayOf(".avi")),
    WOFF("application/x-font-woff", arrayOf(".woff")),
    TFF("application/x-font-ttf", arrayOf(".ttf"))
}